File: leetcode/bfs/111. Minimum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0
        q = deque([root])
        height = 1
        while len(q) != 0:
            size = len(q)
            for _ in range(size):
                node = q.popleft()
                if node.left is None and node.right is None:
                    return height
                if node.left is not None:
                    q.append(node.left)
                if node.right is not None:
                    q.append(node.right)
            height += 1

File: leetcode/111. Minimum Depth of Binary Tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun minDepth(root: TreeNode?): Int {
        var min_depth = Int.MAX_VALUE
        fun dfs(node: TreeNode?, depth: Int) {
            if (node == null) return
            if (depth >= min_depth) return
            if (node.left == null && node.right == null) {
                min_depth = depth
            }
            dfs(node.left, depth+1)
            dfs(node.right, depth+1)
        }
        dfs(root, 1)
        return if (min_depth == Int.MAX_VALUE) return 0 else min_depth
    }
}
